src/provisioner/provisioner.py

```diff
@@ -321,44 +329,58 @@ def _update_provisioning_status(cortx_conf: ConfigStore, node_id: str,
         keys = [(key_prefix + 'phase', phase), (key_prefix + 'status', status)]
         cortx_conf.set_kvs(keys)
 
+    @staticmethod
+    def _is_component_updated(component_name: str, component_version: str):
+        """Verify component version with RELEASE.INFO."""
+        release_version = CortxProvisioner.cortx_release.get_version(component_name)
+        return True if release_version == component_version else False
 
     @staticmethod
     def _validate_provisioning_status(cortx_conf: ConfigStore, node_id: str, apply_phase: str):
         """Validate provisioning."""
         ret_code = 0
         recent_phase = cortx_conf.get(f'node>{node_id}>provisioning>phase')
         recent_status = cortx_conf.get(f'node>{node_id}>provisioning>status')
-        if apply_phase == recent_phase:
-            if recent_status == PROVISIONING_STATUS.PROGRESS.value:
-                Log.info(f'Currently {recent_phase} is in progress, '
-                    f'{apply_phase} is not possible.')
-                return False, ret_code
-            elif (recent_status in [PROVISIONING_STATUS.DEFAULT.value,
-                PROVISIONING_STATUS.ERROR.value]):
-                Log.info(f'{recent_phase} is in {recent_status} state,'
-                    f' {apply_phase} is possible.')
+        if recent_phase is None and recent_status is None:
```

Comment:
instead of several ifs and several similar log can we...
1. log phase, status in the beginning (as it being logged at all the conditions) to avoid duplicate. 
2. Use a table (truth table) and write logic to traverse the truth table. 
    Truth Table can be dict.
    last_phase | last_status | current_phase | code. 
                       |                    |                           |

Reply:
Done, Using dictionary instead multiple if-else statements.